src/client/components/forms/userCollection.js

```diff
@@ -0,0 +1,306 @@
+		const entityType = this.entityType.getValue();
+
+		const data = {
+			collaborators,
+			description,
+			entityType,
+			name,
+			privacy
+		};
+
+		let submissionURL;
+		if (this.props.collection.id) {
+			submissionURL = `/collection/${this.props.collection.id}/edit/handler`;
+		}
+		else {
+			submissionURL = '/collection/create/handler';
+		}
+		request.post(submissionURL)
+			.send(data)
+			.promise()
+			.then((res) => {
+				if (res.status === 200) {
+					window.location.href = `/collection/${res.body.id}`;
+				}
+				else {
+					this.setState(prevState => ({
+						collection: prevState.collection,
+						errorText: `Internal Error: ${res.body.error}`
+					}));
+				}
+			});
+	}
+
+	isValid() {
+		return this.name.getValue() && this.entityType.getValue();
+	}
+
+	getCleanedCollaborators() {
+		const cleanedCollaborators = this.state.collection.collaborators.filter(collaborator => collaborator && collaborator.id !== null);
+		return cleanedCollaborators;
+	}
+
+	handleAddCollaborator() {
+		const newCollaborators = this.state.collection.collaborators;
+		newCollaborators.push({
+			id: null,
+			name: ''
+		});
+		this.setState(prevState => ({
+			collection: {
+				...prevState.collection,
+				collaborators: newCollaborators
+			},
+			errorText: prevState.errorText
+		}));
+	}
+
+	handleRemoveCollaborator(index) {
+		const newCollaborators = this.state.collection.collaborators;
+		newCollaborators.splice(index, 1);
+		this.setState(prevState => ({
+			collection: {
+				...prevState.collection,
+				collaborators: newCollaborators
+			},
+			errorText: prevState.errorText
+		}));
+	}
+
+	handleChangeCollaborator(newCollab, index) {
+		let newCollaborator;
+		if (!newCollab) {
+			newCollaborator = {
+				id: null,
+				name: ''
+			};
+		}
+		else {
+			newCollaborator = newCollab;
+		}
+		const newCollaborators = this.state.collection.collaborators;
+		newCollaborators[index] = newCollaborator;
+		this.setState(prevState => ({
+			collection: {
+				...prevState.collection,
+				collaborators: newCollaborators
+			},
+			errorText: prevState.errorText
+		}));
+	}
+
+	render() {
+		if (this.state.collection.collaborators.length === 0) {
+			this.handleAddCollaborator();
+		}
+
+		const privacyOptions = ['Private', 'Public'].map((option) => ({
+			name: option
+		}));
+		const entityTypeOptions = ['Author', 'Work', 'Edition', 'Edition-Group', 'Publisher'].map((entity) => ({
+			name: entity
+		}));
+		const initialName = this.state.collection.name;
+		const initialDescription = this.state.collection.description;
+		const initialPrivacy = this.state.collection.public ? 'Public' : 'Private';
+		const initialType = this.state.collection.entityType;
+		const {errorText} = this.state;
+		const errorAlertClass =
+			classNames('text-center', 'margin-top-1', {hidden: !errorText});
+		const submitLabel = this.props.collection.name ? 'UPDATE' : 'CREATE';
+
+		/* eslint-disable react/jsx-no-bind */
+		return (
+			<Grid>
```

Comment:
The page layout is a bit off. Instead of a Grid -> Row -> Col hierarchy, try div -> Col only
The Grid adds a 'container' class which throws off the layout in some screen sizes.

I find that there is no need for Row + Col and just having your Col elements a level higher works better (I'm attaching a screenshot where you can see the flattened hierarchy
<img width="1211" alt="Capture d’écran 2020-06-03 à 15 46 59" src="https://user-images.githubusercontent.com/6179856/83644385-878a3e80-a5b1-11ea-97f1-a7476a7847bf.png">